### ISSUE TITLE
How do you prepare those paper_weights (hdf5 files), I tried train.py and got some weights (hdf5 files) but then when I run predict.py nothing shows up (screenshot in description) 

### ISSUE THREAD

**Issue by SabirKhanAkash** [razeineldin/DeepSeg#6]
So the [paper_weights](https://drive.google.com/file/d/1oJQyfe5NHrym9Qk7Wg4nb3_Gvcf97tAN/view) folder that you have provided , contains the weights in hdf5 format for each single model a single time. I trained the model by myself using the **train.py** as I set epoch on 20 so I got 20 HDF5 files in my weights directory. Can you tell me how can I combine them like your [demo paper_weights](https://ibb.co/s2Qn5BJ) folder? I'm attaching a [screenshot ](https://ibb.co/6w4VM5s)for better illustration. Thank You. @razeineldin 

Update: So analyzing ur code I understood that if you are training for like 20 epochs and the code only loads the 20th hdf5 file from weights directory.

> model.py

```
if load_model:
        if config['load_model_path'] is None:
            model.load_weights(glob.glob(os.path.join(config['weight_dir']+config['project_name'],'*'+config['model_num']+'*'))[0])
        else:
            model.load_weights(config['load_model_path'])
    return model
```

But I tried training the model with 
> config['epochs'] = 20 
> config['steps_per_epoch'] = 50 
> config['validation_steps'] = 50

but after the training, I loaded the model with my new weights (20th hdf5 file from weights directory) and I ran the predict.py and I got this [output](https://ibb.co/GChFWJ4) but when I try with your paper weights I get this [correct output](https://ibb.co/Q8Gp5mf). It will be very helpful if you kindly provide the config.py by which you train & obtain your [paper_weights](https://drive.google.com/file/d/1oJQyfe5NHrym9Qk7Wg4nb3_Gvcf97tAN/view) Please help me out.

**Comment by razeineldin** [comment]
Hi @SabirKhanAkash,

Thanks for your question. 
As you already mentioned, only the best weights are used for the prediction. No ensembling or averaging techniques are used.

Regarding the results, I think the training is biased and cannot detect the tumor correctly. Check the training dataset and the loss function. 

Good luck!

**Comment by SabirKhanAkash** [comment]
Thanks @razeineldin for your reply, It seems my generated weights after training is not giving prediction output image like you. I think something is wrong in my train.py . Can you please clarify in which case **config[validate]** should be **True** and when it should be **False** ? which one did u use ? 
I run the train.py with 

```
config[validate] = False 
config['epochs'] = 50
config['steps_per_epoch'] = 250
config['validation_steps'] = 200
```

### my training dataset folders containing 38943 FLAIR images and 38943 truth_segmentation images
### and validation dataset folders containing 12982 FLAIR images and 12982 truth_segmentation images

## you can check my loss fuction after training from [here](https://pastebin.ubuntu.com/p/m7QMb6YdwN/)
Does this repo's config.py is updated according to your paper_weights?

**Comment by SabirKhanAkash** [comment]
So the interesting thing is when I switched 
from 
    TensorFlow 1.15.0 | TensorFlow-gpu 1.15.0 | Keras 2.2 | CUDA 10.0 | cuDNN 7.6
to
    TensorFlow 2.6.0 | TensorFlow-gpu 2.6.0 | Keras 2.6 | CUDA 11.2 | cuDNN 8.1
and modifying some deprecated syntax in the code, now I'm getting some good [predicted output images](https://ibb.co/5KzFLLB) from training.
